addons/edi_stock/models/edi_pick_report_document.py

```diff
@@ -153,6 +153,7 @@ def prepare(self, doc):
         Move = self.env['stock.move']
         # Lock pickings to prevent concurrent report generation attempts
         picks = Picking.search(self.pick_report_domain(doc), order='id')
+        picks = self.filter_picks_with_issues(picks, doc)
```

Comment:
I presume it was not possible to override the pick_report_domain here to return the picks, is it because mapping between failed attempts is difficult, and you can not just look at the pick domain to filter out those records?

Reply:
I tried to override pick_report_domain, didnt find a way better and safer that the one which tbh the performance is lower like this.